src/main/scala/com/wavesplatform/state/diffs/CommonValidation.scala

```diff
@@ -80,7 +80,9 @@ object CommonValidation {
                                              height: Int,
                                              tx: T): Either[ValidationError, T] = tx match {
     case _: PaymentTransaction => Right(tx)
-    case _                     => if (blockchain.containsTransaction(tx.id())) Left(AlreadyInTheState(tx.id(), 0)) else Right(tx)
+    case _ =>
+      val id = tx.id()
+      Either.cond(!blockchain.containsTransaction(id), tx, AlreadyInTheState(id, blockchain.transactionInfo(id).get._1))
```

Comment:
Why not `transactionHeight(id: ByteStr): Option[Int]`? with this, no transaction will ever be loaded